### ISSUE TITLE
CLI argocd app sync APPNAME -l foo=bar

### ISSUE THREAD

**Issue by davidcv5** [argoproj/argo-cd#1241]
Add `argocd app sync APPNAME -l foo=bar` to the CLI

For helm charts composed of multiple micro-services (sub-charts aliases), it would make sense to have a way to do a partial sync of a particular sub-chart without having to specify each resource of the subchart, syncing by resource labels would do the job.

An example use case would be an Application chart tracked by ArgoCD containing multiple sub-charts with their own repo that get built on git push and update the main Application chart, so only those specific resources would need syncing.

This would be particularly helpful for a development environment, where there might be custom modifications in the cluster for testing purposes, not in the application git repo, so a full sync or auto-sync when a micro-service gets updated would override those custom/temporary changes.
Thanks!

**Comment by davidcv5** [comment]
Hello @jessesuen .. After using `argocd` on a daily basis as part of our company CI/CD pipelines for all environments from dev, all the way to prod, here are a few more thoughts I just put in [slack](https://argoproj.slack.com/archives/CASHNF6MS/p1553740299016200) related to partial syncs.

1- Add `argocd app sync APPNAME -l foo=bar` to the CLI (requested above)
2- `argocd app sync --resources=...` should only consider those specific resources. Partial syncs are currently slow, they appear to check the whole application and the output shows the status of the whole application instead of just the resources specified
3- `argocd app sync --resources=XYZ` shouldn't block `argocd app sync --resources=ABC`
4- `argocd app sync --resources=XYZ` should have an option to show only the output for those resources, right now I'm having to use some "hacky" `awk` to parse the response to send some slack notifications
5- `argocd app sync` even though it's supposed to just push the changes, appears to wait for them to be applied, for both full and partial syncs, would be great to have an option that works like a `helm upgrade`, where you just care that spec got to the cluster and get an output immediately

I currently have a helm chart composed of 40+ subchart aliases that form a full application composed of multiple microservices. Each microservice has its own repo, so they individually update the main chart and call `argocd app sync --resource=XYZ` on every build for lower environments.

Calling `argocd app wait` before a partial sync is not an option since it currently waits for any running `sync` to finish instead of those particular resources. So if 5 teams push at the same time for 5 different micro-services, the poor last team would have to wait a lot :slightly_smiling_face:

So in summary, everything boils down to allowing partial syncs to coexist and both `sync` and `wait` should allow partial resources, `sync` already does, though it outputs the whole application status which can be slow.

What do you guys think?
Once again, thanks a lot for building such a great tool!

**Comment by alexmt** [comment]
Implemented, thanks to @simster7 !
